python/cudf/cudf/dataframe/index.py

```diff
@@ -381,6 +384,9 @@ def __init__(self, start, stop=None, name=None):
         self.name = name
         self._cached_values = None
 
+    def __contains__(self, item):
+        return item in self._values
```

Comment:
You shouldn't have to materialize the column for a RangeIndex, you should be able to just check if item is between `self._start` and `self._stop`